### DISCUSSION TITLE
Tracking maximum value for a variable over time

### DISCUSSION THREAD

**Discussion by skv94** [idaholab/moose#22104] [Q&A Tools] [answered]
Hello,
I am simulating a coupled reaction-diffusion system that results in temperature spikes. I am interested in tracking the maximum temperature for the entire domain over the course of the simulation in order to generate a maximum temperature contour. I am currently doing this by generating output files, reading them in Paraview, saving the results as csv files, and then using a postprocessing script in MATLAB to generate the contour as showed below.
![max_temp_2_cm](https://user-images.githubusercontent.com/78758548/190386891-eec0579d-7e0c-46ff-954a-56a2c70368ad.jpg)
However, this is quite time consuming and I was wondering if it is possible to use Postprocessors to keep track of the maximum temperature over all time steps. I tried to use a combination of the 'NodalExtremeValue' and 'TimeExtremeValue' postprocessors, but that simply outputs a scalar corresponding to the node with the highest temperature for each time step.

Can I please get some advice?

Thanks in advance.

**Comment by hugary1995** [comment, marked as answer]
You could use a combination of auxkernels to achieve this

```python
[AuxVariables]
  [T_old]
    [AuxKernel]
      type = ParsedAux
      args = 'T'
      function = 'T'
      execute_on = 'INITIAL TIMESTEP_END'
    []
  []
  [T_max]
    [AuxKernel]
      type = ParsedAux
      args = 'T T_old'
      function = 'if(T>T_old, T, T_old)'
      execute_on = 'LINEAR'
    []
  []
[]
```
The trick is in the execute_on flags.

**Reply by GiudGiud** [reply]
same trick with a ParsedPostprocessor should work

**Reply by hugary1995** [reply]
ya, although I guess OP was asking for something like what's shown in the image. Anyways...
